### ISSUE TITLE
Organization deployment keys

### ISSUE THREAD

**Issue by jolting** [isaacs/github#500]
It seems strange that organization webhooks exist, but organization deployment keys do not.

Suppose I want to use a push event to deploy something using a private organization. The deployment server needs a deployment key for read-access. I believe it is necessary to set a deployment key for each repo. It would be nice to set one key for all private repositories in an organization.

**Comment by benatkin** [comment]
I bit the bullet and created a machine user.

**Comment by jolting** [comment]
Good point. Take the travisbot approach. I'll try that.

**Comment by VelizarVESSELINOV** [comment]
Today, Continuous Integration of several private GitHub projects it is very challenging, we need to create a fake user account as described in: https://docs.travis-ci.com/user/private-dependencies/#Dedicated-User-Account
It would be great if GitHub proposes this solution by default instead of creating fake accounts with extra complexity.

**Comment by adamkdean** [comment]
This is still an issue. It would be fair enough if we could create a deploy key and use it multiple times but we cannot. It means for a lot of projects, we end up having lots of keys. Or a machine user.

**Comment by amenadiel** [comment]
Yup, and machine users count as seats, so you actually pay to have a simple read-only bot

**Comment by nelsonjchen** [comment]
I've noticed that these days some CI services can be granted access to organizations and sets of repos in the organization. Could something like that be used ad-hoc to make something that might satisfy this or most of this?

**Comment by Burnett01** [comment]
Another option might be to utilize Githubs API. https://developer.github.com/v3/repos/keys/#add-a-new-deploy-key

**Comment by benbonnet** [comment]
@Burnett01 the link you provide let you add a deploy key on a repo, not on an org - which is not the issue here if I'm not wrong; or I'm might be missing something

**Comment by Burnett01** [comment]
> @Burnett01 the link you provide let you add a deploy key on a repo, not on an org - which is not the issue here if I'm not wrong; or I'm might be missing something

We ended up adding deployment keys to all org repositories by using the API, as described in my previous comment. Nowadays this is actually no longer needed thanks to Github Actions and its orga-wide secrets. Unless of course, one does not want to use actions for CI/CD, the best way is to resort to using the API. Our orga Github-App automagically assigns deploy-keys to our repos that need it.

**Comment by digisavvy** [comment]
This is an issue for me. I have an organization and would love to add an SSH key to my org.
